examples/model_selection/plot_grid_search_stats.py

```diff
@@ -0,0 +1,551 @@
+# between the performance of the models. This means that some partitions of the
+# data can make the distinction of the classes particularly easy or hard to
+# find for all models, and thus their scores will co-vary.
+#
+# Let's inspect this partition effect by plotting the performance of all models
+# in each fold, and calculating the correlation between models across folds:
+
+# create df of model scores ordered by perfomance
+model_scores = results_df.filter(like='split')
+
+# plot 30 examples of dependency between cv fold and AUC scores
+fig, ax = plt.subplots()
+sns.lineplot(
+    data=model_scores.transpose().iloc[:30],
+    dashes=False, palette='Set1', marker='o', alpha=.5, ax=ax
+)
+ax.set_xlabel("CV test fold", size=12, labelpad=10)
+ax.set_ylabel("Model AUC", size=12)
+ax.tick_params(bottom=True, labelbottom=False)
+plt.show()
+
+# print correlation of AUC scores across folds
+print(f"Correlation of models:\n {model_scores.transpose().corr()}")
+
+# %%
+# We can observe that the performance of the models highly depends on the fold.
+#
+# As a consequence, if we assume independence between observations we will be
+# underestimating the variance computed in our statistical tests, increasing
+# the number of false positive errors (i.e. detecting a significant difference
+# between models when such does not exist) [1]_.
+#
+# Several methods have been developed to correct the variance for the
+# correlation in these cases. In this example we will explore one of these
+# methods implemented using two different statistical approaches: frequentist
+# and Bayesian.
+
+# %%
+# Comparing two models: frequentist approach
+# ------------------------------------------
+#
+# We can start by asking: "Is the first model significantly better than the
+# second model (when ranked by `mean_test_score`)?"
+#
+# To answer this question using a frequentist approach we could
+# run a paired t-test and compute a p-value. Many variants of the latter have
+# been developed to account for the 'non-independence of observations problem'
+# described in the previous section. We will use the one proven to obtain the
+# highest replicability scores (which rate how similar the performance of a
+# model is when evaluating it on different random partitions of the same
+# dataset) while mantaining a low rate of false postitives and false negatives:
+# the Nadeau and Bengio's corrected t-test [2]_ that uses a 10 times repeated
+# 10-fold cross validation [3]_.
+#
+# This corrected paired t-test is computed as:
+#
+# .. math::
+#    t=\frac{\frac{1}{k \cdot r}\sum_{i=1}^{k}\sum_{j=1}^{r}x_{ij}}
+#    {\sqrt{(\frac{1}{k \cdot r}+\frac{n_2}{n_1})\hat{\sigma}^2}}
+#
+# where :math:`{\sigma}^2` represents the variance, :math:`k` is the number of
+# folds, :math:`r` the number of repetitions in the cross-validation,
+# :math:`n_2` is the number of observations used for testing, and :math:`n_1`
+# is the the number of observations used for training.
```

Comment:
Maybe rename `n_1` into `n_{\mathrm{train}}` and `n_2` into `n_{\mathrm{test}}`.